classes/scheduledTask/ScheduledTask.php

```diff
@@ -157,6 +152,33 @@ public function execute(): bool
 
         return $result;
     }
+
+    /**
+     * Write an entry into the execution log.
+     *
+     * @param string $logFile Path to the log file.
+     * @param string $message A translated message.
+     * @param ?string $type One of the ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE... constants
+     */
+    public static function writeToExecutionLogFile(string $logFile, string $message, ?string $type = null): void
```

Comment:
as we do not have any good logging system and this task dispatch several jobs, we need a way to write log to task's log file . We can pass the task instance but that introduce a possible chances to handle closure in future . This static interface and new method `getExecutionLogFile` allow a mechanism to write logs to task's log file even after task complete and control passed to jobs for better log centralization for specific task .